Makefile

```diff
@@ -152,6 +165,25 @@ endif
 	rm -f LICRES
 	@echo "  all OK"
 
+# Check licenses of go dependencies
+.PHONY: go-license-check
+go-license-check: $(GO_LICENSES_BIN)
```

Comment:
Should CI include this check?

Reply:
> It's really only needed when dependencies change.

that makes sense to me. We can leverage dorny/paths-filter to run `go-license-check` when the `go.mod` gets changed.

for another, maybe we can add it to https://yunikorn.apache.org/community/release_procedure to remind RM that `go-license-check` must pass before release.

Reply:
Adding to the release procedure is definitely on the roadmap. This is a first step, to give us a way to track the licenses in the first place.